# Q2: 03-Aug-2022

Please refer to the same Universities.xlsx on GitHub (https://github.com/Accelerate-AI/UnsupervisedLearning_Clustering/tree/main/DailyQuestions/Universities.xlsx). 
Find out the Manhattan distance between Stanford and MIT.

What difference you can convey for Manhattan distance w.r.t. Euclidean distance measure?

In [1]:
import pandas as pd
import numpy as np

In [2]:
col_list = ['Univ', 'SAT', 'Top10', 'Accept', 'SFRatio', 'Expenses', 'GradRate']
df = pd.read_excel("Universities.xlsx", usecols=col_list)
df.head(2)

Univ   SAT  Top10  Accept  SFRatio  Expenses  GradRate
0    Brown  1310     89      22       13     22704        94
1  CalTech  1415    100      25        6     63575        81

In [3]:
series_Stanford = df[df['Univ'] == "Stanford"].copy().drop('Univ', axis = 1).iloc[0]
series_MIT = df[df['Univ'] == "MIT"].copy().drop('Univ', axis = 1).iloc[0]

In [4]:
def get_manhattan_distince(series_1, series_2):
    return np.round(np.sum(np.abs(series_1 - series_2)), 2)

In [5]:
mh_dist = get_manhattan_distince(series_Stanford, series_MIT)
print(f"Manhattan distance is : {mh_dist}")

Manhattan distance is : 1618


In [6]:
# series_Stanford
# series_MIT
#np.abs(series_MIT - series_Stanford)
#(series_Stanford - series_MIT) ** 2

# Answer:
    1. The Manhattan distance between Stanford and MIT is 1618.
    2. 
        i.  If we only look at current data, we can say that both Manhattan and Euclidean, methods suffer when 
            one feature value is very large as compared to others features. Here too, we see that overall Manhattan 
            distance is 1618, out of which 1580 comes due to the difference between Expenses.
        ii. The link (https://datascience.stackexchange.com/questions/20075/when-would-one-use-manhattan-distance-as-opposed-to-euclidean-distance), 
            explains the difference. Here its explained that for higher dimensional data, Manhattan distance will perform 
            better than Euclidean distance.